### DISCUSSION TITLE
Is there a way to dynamically change the lov parameter of a selector control?

### DISCUSSION THREAD

**Discussion by dennis31197** [Avaiga/taipy#953] [Q&A]
I want to create a dropdown with certain dates from a range. 
That range should be defined by a date_range control. 
So I need to dynamically change the possible dropdown values based on the selcted dates in the date_range control.

Basic example would be below. The state.selected_datetime updates directly without any problem.
Only the state.datetime_str seems to have no effect. Even after refreshing the page. Thanks for any help.

```python
df = pd.DataFrame(...)
def extract_datetime(...):
    ...
def select_datetime_change(state, var_name, var_value):
    if var_name == "datetime_range":
        datetime_values = extract_datetime(df, var_value[0], var_value[1])
        state.datetime_str = ';'.join(datetime_values.tolist())
        state.selected_datetime = datetime_values[0]
            
with tgb.part():
    tgb.text("Select Time Window with Time:")
    tgb.date_range("{datetime_range}", with_time=True, on_change=select_datetime_change)
    tgb.selector(value="{selected_datetime}", lov=datetime_str, dropdown=True, label="Selected Datetime")
```

**Comment by FlorianJacta** [comment]
I see you have closed the issue. You may have found the solution. For future reference, here is a possible solution. To tell Taipy that a variable is interactive, you have to use its interactive representation:

`'{your_variable}'`

This variable reflects its real-time value and is interactive (can be changed in real-time). The code below should work:

```python
          
with tgb.part():
    ...
    tgb.selector(value="{selected_datetime}", lov="{datetime_str}", dropdown=True, label="Selected Datetime")
```
